assets/src/block-templates/register.js

```diff
@@ -34,7 +40,29 @@ export const registerBlockTemplates = blockTemplates => {
 
     registerBlockType(metadata, {
       edit: edit(template, templateLock),
-      save,
+      save: props => {
+        if(props.innerBlocks.length) {
+          const className = `${props.innerBlocks[0].attributes.className || ''}`;
+          let patternClassName = '';
+
+          switch (metadata.type) {
```

Comment:
Instead of a switch on type your could declare/fix the `category` property in `block.json` and use that category in a classname.

Reply:
Yeah! That was the initial idea, but in that case we'll need to update all of `block.json` files, not only the one related to campaigns.

Reply:
right, but I don't think it would be an issue since those declarations don't appear in the editor block list (cf l.16), nothing should change visually in the editor ?